llama_index/evaluation/retrieval/metrics.py

```diff
@@ -0,0 +1,64 @@
+from llama_index.evaluation.retrieval.metrics_base import BaseRetrievalMetric, RetrievalMetricResult
+from abc import abstractmethod, ABC
+from typing import List, Any, Dict, Optional
+from pydantic import BaseModel, Field
+
+
+class HitRate(BaseRetrievalMetric):
+    """Hit rate metric."""
+
+    metric_name: str = "hit_rate"
+
+    def compute(
+        self,
+        query: str,
+        expected_ids: List[str],
+        retrieved_ids: List[str],
+        **kwargs: Any
+    ) -> RetrievalMetricResult:
+        """Compute metric."""
+        if retrieved_ids is None or expected_ids is None:
+            raise ValueError("Retrieved ids and expected ids must be provided")
+        return RetrievalMetricResult(
+            score=float(any([id in expected_ids for id in retrieved_ids])),
+        )
+
+
+class MRR(BaseRetrievalMetric):
+    """MRR metric."""
+
+    metric_name: str = "mrr"
+    
+    def compute(
+        self,
+        query: Optional[str] = None,
+        expected_ids: Optional[List[str]] = None,
+        retrieved_ids: Optional[List[str]] = None,
+        **kwargs: Any
+    ) -> RetrievalMetricResult:
+        """Compute metric."""
+        if retrieved_ids is None or expected_ids is None:
+            raise ValueError("Retrieved ids and expected ids must be provided")
+        for i, id in enumerate(retrieved_ids):
+            if id in expected_ids:
+                return RetrievalMetricResult(
+                    score=1.0 / (i + 1),
+                )
+        return RetrievalMetricResult(
+            score=0.0,
+        )
+
+
+METRIC_REGISTRY: Dict[str, BaseRetrievalMetric] = {
+    "hit_rate": HitRate(),
+    "mrr": MRR(),
```

Comment:
Might be interesting to add ROUGE as a metric (token overlap between retrieved and expected context) or even raw similarity measures (BertScore?)

Reply:
Maybe in a future PR though

Reply:
yeah definitely!